doc/modules/preprocessing.rst

```diff
@@ -593,6 +593,64 @@ the 2 features with respectively 3 categories each.
 See :ref:`dict_feature_extraction` for categorical features that are
 represented as a dict, not as scalars.
 
+.. _one_hot_encoder_infrequent_categories:
+
+Infrequent categories
+---------------------
+
+:class:`OneHotEncoder` supports outputing a feature that combines infrequent
+categories in the training data. For each input feature that has an infrequent
+category a new column is formed to represent it. The parameters to enable the
+gathering of infrequent categories are `min_frequency` and `max_categories`.
+
+1. `min_frequency` is either an  integer greater or equal to 1, or a float in
+the interval `(0.0, 1.0)`. If `min_frequency` is an integer, categories with a
+cardinality smaller than `min_frequency * n_samples`  will be considered
+infrequent. If `min_frequency` is a float, categories with a cardinality smaller
+than this fraction of the total number of samples will be considered infrequent.
+
+2. `max_categories` is either `None` or any integer greater than 1. This
+parameter sets an upper limit to the number of output features for each input
+feature. `max_categories` includes the feature that combines infrequent
+categories.
+
+In the following example, the categories, `'dog', 'snake'` are considered
+infrequent::
+
+   >>> X = np.array([['dog'] * 5 + ['cat'] * 20 + ['rabbit'] * 10 + 
+   ...               ['snake'] * 3]).T
+   >>> enc = preprocessing.OneHotEncoder(min_frequency=6,
+   ...                                   handle_unknown='auto').fit(X)
```

Comment:
is the handle_unknown here to not show a deprecation warning?

Reply:
The default is `error`, which doesn't handle unknown well.

I most likely need to add testing for infrequent categories with `handle_unknown='error'`.

Reply:
Although, I would suspect it to be not that useful to have.